File: frontend/src/infrastructure/api/reports.ts
import { api } from '@/lib/api';

export const reportsApi = {
  generateClientReport: (interviewId: string, token: string) =>
    api.get<{ markdown: string }>(`/reports/client/${interviewId}`, token),

  generateAgentReport: (interviewId: string, token: string) =>
    api.get<{ markdown: string }>(`/reports/agent/${interviewId}`, token),
};

File: backend/app/api/v1/endpoints/reports.py
from uuid import UUID

from fastapi import APIRouter, HTTPException

from app.application.services.report_service import ReportService
from app.core.deps import CurrentUser

router = APIRouter()


@router.get("/client/{interview_id}")
async def generate_client_report(interview_id: UUID, _: CurrentUser):
    service = ReportService()
    markdown = await service.generate_client_report(interview_id)
    if not markdown:
        raise HTTPException(status_code=404, detail="Interview not found")
    return {"markdown": markdown}


@router.get("/agent/{interview_id}")
async def generate_agent_report(interview_id: UUID, _: CurrentUser):
    service = ReportService()
    markdown = await service.generate_agent_report(interview_id)
    if not markdown:
        raise HTTPException(status_code=404, detail="Interview not found")
    return {"markdown": markdown}

